pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -1304,6 +1357,305 @@ def plot_new_spend_contributions(
+        also highlighted.
+
+        Parameters
+        ----------
+        x : np.ndarray
+            The x-axis data, usually representing the amount of
+            input (e.g., substrate concentration in enzymology terms).
+        ax : plt.Axes
+            The matplotlib axes object where the plot should be drawn.
+        channel : str
+            The name of the channel for which the curve fit is being plotted.
+        color_index : int
+            An index used for color selection to ensure distinct colors for multiple plots.
+        xlim_max: int
+            The maximum value to be plot on the X-axis
+        method: str
+            The method used to fit the contribution & spent non-linear relationship.
+            It can be either 'sigmoid' or 'michaelis-menten'.
+
+        Returns
+        -------
+        None
+            The function modifies the given axes object in-place and doesn't return any object.
+        """
+
+        if self.X is not None:
+            x_mean = np.max(self.X[channel])
+
+        # Set x_limit based on the method or xlim_max
+        if xlim_max is not None:
+            x_limit = xlim_max
+        else:
+            x_limit = x_mean
+
+        # Generate x_fit and y_fit
+        x_fit = np.linspace(0, x_limit, 1000)
+        upper_params = self.format_recovered_transformation_parameters(
+            quantile=quantile_upper
+        )
+        lower_params = self.format_recovered_transformation_parameters(
+            quantile=quantile_lower
+        )
+        mid_params = self.format_recovered_transformation_parameters(quantile=0.5)
+        y_fit = self.saturation.function(
+            x=x_fit, **mid_params[channel]["saturation_params"]
+        ).eval()
+
+        y_fit_lower = self.saturation.function(
+            x=x_fit, **lower_params[channel]["saturation_params"]
+        ).eval()
+        y_fit_upper = self.saturation.function(
+            x=x_fit, **upper_params[channel]["saturation_params"]
+        ).eval()
+
+        # scale all y fit values to the original scale using
+        # `mmm.target_transformer.named_steps["scaler"].scale_.item()`
+        y_fit = (
+            self.get_target_transformer()
+            .inverse_transform(y_fit.reshape(-1, 1))
+            .flatten()
+        )
+        y_fit_lower = (
+            self.get_target_transformer()
+            .inverse_transform(y_fit_lower.reshape(-1, 1))
+            .flatten()
+        )
+        y_fit_upper = (
+            self.get_target_transformer()
+            .inverse_transform(y_fit_upper.reshape(-1, 1))
+            .flatten()
+        )
+
+        # scale x fit values
+        x_fit = self._channel_map_scales()[channel] * x_fit
+
+        ax.fill_between(
+            x_fit, y_fit_lower, y_fit_upper, color=f"C{color_index}", alpha=0.25
+        )
+        ax.plot(x_fit, y_fit, color=f"C{color_index}", label=label, alpha=0.6)
+
+        ax.set(xlabel="Spent", ylabel="Contribution")
+        ax.legend()
```

Comment:
Can you return the `fig` object (and add it to the type hints and docstrings)?

Reply:
This function is like a helper.

```python
Returns
        -------
        None
            The function modifies the given axes object in-place and doesn't return any object.
```

or what you mean? The main function returns a figure.